src/overmap.cpp

```diff
@@ -782,7 +782,14 @@ void overmap_special::finalize()
         if( !elem.terrain && oter.terrain ) {
             elem.terrain = oter.terrain->get_type_id();    // Defaulted.
         }
-        elem.connection = overmap_connections::guess_for( elem.terrain );
+
+        // If the connection type hasn't been specified, we'll guess for them.
+        // The guess isn't always right (hence guessing) in the case where
+        // multiple connections types can be made on a single location type,
+        // e.g. both roads and forest trails can be placed on "forest" locations.
+        if( elem.connection == string_id<overmap_connection>::NULL_ID() ) {
```

Comment:
Instead of comparing with the null-id, you can call `is_null` on the id to check whether it's the null-id (that's a bit shorter).